src/ext/ddtrace.c

```diff
@@ -795,13 +807,169 @@ static PHP_FUNCTION(dd_tracer_circuit_breaker_info) {
+    };
+    if (ddtrace_string_equals(subject, str_1) || ddtrace_string_equals(subject, str_true)) {
+        return true;
+    }
+    ddtrace_string str_0 = {
+        .ptr = "0",
+        .len = 1,
+    };
+    ddtrace_string str_false = {
+        .ptr = "false",
+        .len = sizeof("false") - 1,
+    };
+    if (ddtrace_string_equals(subject, str_0) || ddtrace_string_equals(subject, str_false)) {
+        return false;
+    }
+    return default_value;
+}
+
+static bool _dd_config_trace_enabled() {
+    char *value = getenv("DD_TRACE_ENABLED");
+    ddtrace_zppstrlen_t value_len;
+    if (value && (value_len = strlen(value))) {
+        zval item;
+#if PHP_VERSION_ID < 70000
+        ZVAL_STRINGL(&item, value, value_len, 1);
+#else
+        ZVAL_STRINGL(&item, value, value_len);
+#endif
+        bool result = _dd_config_bool(&item, true);
+        ddtrace_zval_ptr_dtor(&item);
+        return result;
+    } else {
+        return true;
+    }
+}
+
+static PHP_FUNCTION(ddtrace_config_trace_enabled) {
+    PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht TSRMLS_CC);
+    PHP7_UNUSED(INTERNAL_FUNCTION_PARAM_PASSTHRU);
+    RETURN_BOOL(_dd_config_trace_enabled());
+}
+
+/* Annoyingly, zend_memnstr changed from char* to const char* in PHP 5.6, but
+ * not for `end`! That happened in 7.
+ * We aren't modifying anything, so we'll use const and just cast inside.
+ */
+static const char *ddtrace_memnstr(const char *haystack, const char *needle, int needle_len, const char *end) {
+#if PHP_VERSION_ID < 50600
+    return zend_memnstr((char *)haystack, (char *)needle, needle_len, (char *)end);
+#elif PHP_VERSION_ID < 70000
+    return zend_memnstr((const char *)haystack, (const char *)needle, needle_len, (char *)end);
+#else
+    return zend_memnstr((const char *)haystack, (const char *)needle, needle_len, (const char *)end);
+#endif
+}
+
+// note: only call this if _dd_config_trace_enabled() returns true
+static bool _dd_config_integration_enabled(ddtrace_string integration) {
+    char *integrations_disabled = getenv("DD_INTEGRATIONS_DISABLED");
+    ddtrace_zppstrlen_t integrations_disabled_len;
+    if (integrations_disabled && (integrations_disabled_len = strlen(integrations_disabled))) {
+        // todo: cache this into an array with string keys?
+        const char *begin = integrations_disabled, *end = integrations_disabled + integrations_disabled_len;
+        const char *found;
+    loop:
+        found = ddtrace_memnstr(begin, integration.ptr, integration.len, end);
+        if (found) {
+            bool valid_begin = found == begin || *(found - 1) == ',';
+            const char *found_end = found + integration.len;
+            if (valid_begin && (found_end == end || *found_end == ',')) {
+                return false;
+            } else {
+                // todo: remove goto?
+                begin = found + 1;
+                goto loop;
+            }
+        }
+    }
+    return true;
```

Comment:
Do you think we could make [`find_exact_match()` public](https://github.com/DataDog/dd-trace-php/blob/master/src/ext/request_hooks.c#L16-L34) and use it here instead? Or combine them somehow since they are essentially doing the same thing? Or at least very similar? :)

Reply:
We should definitely combine them. Thanks for bringing this to my attention!